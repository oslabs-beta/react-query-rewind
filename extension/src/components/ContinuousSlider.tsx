import * as React from 'react';
import Box from '@mui/material/Box';
import Stack from '@mui/material/Stack';
import Slider from '@mui/material/Slider';

type ContinuousSliderProps = {
  value: number;
  maxValue: number;
  onChange: (newValue: number) => void;
};

export default function ContinuousSlider({
  value,
  maxValue,
  onChange,
}: ContinuousSliderProps) {
  const handleChange = (event: Event, newValue: number | number[]) => {
    onChange(newValue as number);
  };

  return (
        <Slider
          value={value}
          onChange={handleChange}
          max={maxValue}
          aria-labelledby='continuous-slider'
        />
  );
}
