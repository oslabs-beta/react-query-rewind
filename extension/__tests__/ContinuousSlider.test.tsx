import * as React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ContinuousSlider from '../src/components/ContinuousSlider'; // Adjust the import path as necessary
import '@testing-library/jest-dom';

describe('ContinuousSlider', () => {
  it('renders correctly with given props', () => {
    const handleChange = jest.fn();
    const { asFragment } = render(<ContinuousSlider value={30} maxValue={100} onChange={handleChange} />);

    const slider = screen.getByRole('slider');
    expect(slider).toBeInTheDocument();
    expect(slider).toHaveAttribute('aria-valuemax', '100');
    expect(slider).toHaveAttribute('aria-valuenow', '30');
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onChange with the new value when changed', async () => {
    const handleChange = jest.fn();
    render(<ContinuousSlider value={30} maxValue={100} onChange={handleChange} />);

    // Assuming the slider allows keyboard input, you can simulate changes like this
    const slider = screen.getByRole('slider');
    await userEvent.click(slider);
    await userEvent.keyboard('{arrowright}'); // Simulates pressing the right arrow key to increase the value

    expect(handleChange).toHaveBeenCalledWith(expect.any(Number));
  });

});
