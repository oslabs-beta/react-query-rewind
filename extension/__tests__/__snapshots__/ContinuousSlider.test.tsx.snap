// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContinuousSlider renders correctly with given props 1`] = `
<DocumentFragment>
  <span
    class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-187mznn-MuiSlider-root"
  >
    <span
      class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
    />
    <span
      class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
      style="left: 0%; width: 30%;"
    />
    <span
      class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
      data-index="0"
      style="left: 30%;"
    >
      <input
        aria-labelledby="continuous-slider"
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="30"
        data-index="0"
        max="100"
        min="0"
        step="1"
        style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
        type="range"
        value="30"
      />
    </span>
  </span>
</DocumentFragment>
`;
